py/desitarget/mock/build.py

```diff
@@ -1042,3 +1060,509 @@ def join_targets_truth(mockdir, outdir=None, force=False, comm=None):
         mtl.meta['EXTNAME'] = 'MTL'
         mtl.write(tmpout, overwrite=True, format='fits')
         os.rename(tmpout, out_mtl)
+
+def _get_magnitudes_onebrick(specargs):
+    """Filler function for the multiprocessing."""
+    return get_magnitudes_onebrick(*specargs)
+
+def get_magnitudes_onebrick(target_name, mockformat, thisbrick, brick_info, Magnitudes, dust_dir,
+                         select_targets_function, source_data, rand, log):
+    """Wrapper function to generate magnitudes for all the objects on a single brick."""
+
```

Comment:
Speed-up is only one issue.  The bigger issue that my (future) refactor will address is that the target densities don't end up being correct.  For the QSOs the issue is described in #192 in detail.  I don't know whether your *no-spectra* version of `targets_truth` suffers from this problem.

On principle I also think we shouldn't be cutting and pasting dozens of lines of code.  

I should also mention that I don't expect any other major changes to the "generate spectra" version of `targets_truth` after the refactor which fixes the target densities.